src/main.cpp

```diff
@@ -448,6 +449,31 @@ static void parse_arguments(int argc, char** argv, Globals& g) {
             case 'l':
                 g.initial_monitors_locked = 1;
                 break;
+            case 'h':
+                std::cout << "This starts the herbstluftwm window manager. In order to" << endl;
+                std::cout << "interact with a running herbstluftwm instance, use the" << endl;
+                std::cout << "\'herbstclient\' command." << endl;
+                std::cout << endl;
+                std::cout << "The herbstluftwm command accepts the following options:" << endl;
+                std::cout << endl;
+                for (size_t i = 0; long_options[i].name; i++) {
+                    std::cout << "    ";
+                    if (long_options[i].val != 1) {
+                        std::cout << "-" << static_cast<char>(long_options[i].val);
+                        if (long_options[i].has_arg) {
+                            std::cout << " ARG";
+                        }
+                        std::cout << "  ";
```

Comment:
I think `", "` would be a more idiomatic way for listing the short and long form of arguments

Reply:
Besides that, I even tried the PR locally since there's no test (not appropriate, I guess...), and I looks good to me. :+1:

Reply:
I didn't know what to test (beside that running it does not crash). If you have an idea, I can also add a testcase.
I've replaced the separator by `", "`. (I didn't know which command to check for comparison; but `cp` indeed uses `", "` and it looks good):

```
$ ./herbstluftwm -h
This starts the herbstluftwm window manager. In order to
interact with a running herbstluftwm instance, use the
'herbstclient' command.

The herbstluftwm command accepts the following options:

    -v, --version
    -h, --help
    -c ARG, --autostart ARG
    -l, --locked
    --exit-on-xerror
    --no-tag-import
    --verbose

See the herbstluftwm(1) man page for their meaning.
```

Reply:
Imho, a proper test would just run `herbstluftwm -h` (and `--help`) and compare its output. But as I said, I don't think that's appropriate because it's not really a kind of test that fits very well into the test framework, and it's rather low value. That value is mainly that it would have allowed me to take a look at the expected output without actually running it manually.